WalletWasabi/WabiSabi/Backend/Rounds/CoinJoinStorage/CoinJoinIdStore.cs

```diff
@@ -50,19 +50,19 @@ public override bool Contains(uint256 id)
 	public static CoinJoinIdStore Create(string coinJoinIdStoreFilePath)
 	{
 		bool updateFile = false;
-		var coinjoins = Enumerable.Empty<string>();
+		var coinjoins = new List<string>();
 		if (!File.Exists(coinJoinIdStoreFilePath))
 		{
 			IoHelpers.EnsureFileExists(coinJoinIdStoreFilePath);
 		}
 		else
 		{
-			coinjoins = File.ReadAllLines(coinJoinIdStoreFilePath).Where(line => !string.IsNullOrEmpty(line));
+			coinjoins = File.ReadAllLines(coinJoinIdStoreFilePath).Where(line => !string.IsNullOrEmpty(line)).ToList();
```

Comment:
It keeps the ordering depending on the data structure that implements it. For example, a `HashSet` is an enumerable that doesn't respect the ordering. I used `List` instead of `IEnumerable` to explicitely ask for a data structure. So it's not necessary, but it's to not use the function incorrectly by mistake. I can revert these changes is you think they are better be reverted.

Reply:
Yep the real fix here is to change the `Dictionnary` to `List`. I will implement your suggestions